include/bluetooth/services/ots.h

```diff
@@ -193,6 +196,124 @@ enum {
+
+/** @brief Set @ref BT_OTS_DIR_LIST_FLAG_CUR_SIZE flag.
+ *
+ *  @param flags Directory Listing Object flags.
+ */
+#define BT_OTS_DIR_LIST_SET_FLAG_CUR_SIZE(flags) \
+	WRITE_BIT((flags), BT_OTS_DIR_LIST_FLAG_CUR_SIZE, 1)
+
+/** @brief Set @ref BT_OTS_DIR_LIST_FLAG_ALLOC_SIZE flag.
+ *
+ *  @param flags Directory Listing Object flags.
+ */
+#define BT_OTS_DIR_LIST_SET_FLAG_ALLOC_SIZE(flags) \
+	WRITE_BIT((flags), BT_OTS_DIR_LIST_FLAG_ALLOC_SIZE, 1)
+
+/** @brief Set @ref BT_OTS_DIR_LIST_FLAG_FIRST_CREATED flag.
+ *
+ *  @param flags Directory Listing Object flags.
+ */
+#define BT_OTS_DIR_LIST_SET_FLAG_FIRST_CREATED(flags) \
+	WRITE_BIT((flags), BT_OTS_DIR_LIST_FLAG_FIRST_CREATED, 1)
+
+/** @brief Set @ref BT_OTS_DIR_LIST_FLAG_LAST_MODIFIED flag.
+ *
+ *  @param flags Directory Listing Object flags.
+ */
+#define BT_OTS_DIR_LIST_SET_FLAG_LAST_MODIFIED(flags) \
+	WRITE_BIT((flags), BT_OTS_DIR_LIST_FLAG_LAST_MODIFIED, 1)
+
+/** @brief Set @ref BT_OTS_DIR_LIST_FLAG_PROPERTIES flag.
+ *
+ *  @param flags Directory Listing Object flags.
+ */
+#define BT_OTS_DIR_LIST_SET_FLAG_PROPERTIES(flags) \
+	WRITE_BIT((flags), BT_OTS_DIR_LIST_FLAG_PROPERTIES, 1)
+
+/** @brief Set @ref BT_OTS_DIR_LIST_FLAG_EXTENDED flag.
+ *
+ *  @param flags Directory Listing Object flags.
+ */
+#define BT_OTS_DIR_LIST_SET_FLAG_EXTENDED(flags) \
+	WRITE_BIT((flags), BT_OTS_DIR_LIST_FLAG_EXTENDED, 1)
+
+/** @brief Get @ref BT_OTS_DIR_LIST_GET_FLAG_TYPE_128 flag.
+ *
+ *  @param flags Directory Listing Object flags.
+ */
+#define BT_OTS_DIR_LIST_GET_FLAG_TYPE_128(flags) \
+	((flags) & BIT(BT_OTS_DIR_LIST_FLAG_TYPE_128))
+
+/** @brief Get @ref BT_OTS_DIR_LIST_GET_FLAG_CUR_SIZE flag.
+ *
+ *  @param flags Directory Listing Object flags.
+ */
+#define BT_OTS_DIR_LIST_GET_FLAG_CUR_SIZE(flags) \
+	((flags) & BIT(BT_OTS_DIR_LIST_FLAG_CUR_SIZE))
+
+/** @brief Get @ref BT_OTS_DIR_LIST_GET_FLAG_ALLOC_SIZE flag.
+ *
+ *  @param flags Directory Listing Object flags.
+ */
+#define BT_OTS_DIR_LIST_GET_FLAG_ALLOC_SIZE(flags) \
+	((flags) & BIT(BT_OTS_DIR_LIST_FLAG_ALLOC_SIZE))
+
+/** @brief Get @ref BT_OTS_DIR_LIST_GET_FLAG_FIRST_CREATED flag.
+ *
+ *  @param flags Directory Listing Object flags.
+ */
+#define BT_OTS_DIR_LIST_GET_FLAG_FIRST_CREATED(flags) \
+	((flags) & BIT(BT_OTS_DIR_LIST_FLAG_FIRST_CREATED))
+
+/** @brief Get @ref BT_OTS_DIR_LIST_GET_FLAG_LAST_MODIFIED flag.
+ *
+ *  @param flags Directory Listing Object flags.
+ */
+#define BT_OTS_DIR_LIST_GET_FLAG_LAST_MODIFIED(flags) \
+	((flags) & BIT(BT_OTS_DIR_LIST_FLAG_LAST_MODIFIED))
+
+/** @brief Get @ref BT_OTS_DIR_LIST_GET_FLAG_PROPERTIES flag.
+ *
+ *  @param flags Directory Listing Object flags.
+ */
+#define BT_OTS_DIR_LIST_GET_FLAG_PROPERTIES(flags) \
+	((flags) & BIT(BT_OTS_DIR_LIST_FLAG_PROPERTIES))
+
+/** @brief Get @ref BT_OTS_DIR_LIST_GET_FLAG_EXTENDED flag.
+ *
+ *  @param flags Directory Listing Object flags.
+ */
+#define BT_OTS_DIR_LIST_GET_FLAG_EXTENDED(flags) \
+	((flags) & BIT(BT_OTS_DIR_LIST_FLAG_EXTENDED))
```

Comment:
I don't see any API to configure the Directory Listing object flags. The Directory listing object is created internally based on configuration. Can we move these macros and enums to the ots_dir_list_internal header?

Reply:
Agreed